subprojects/core/src/test/groovy/org/gradle/api/internal/plugins/ExtensionsStorageTest.groovy

```diff
@@ -63,6 +63,15 @@ class ExtensionsStorageTest extends Specification {
         thrown UnknownDomainObjectException
     }
 
+    def "configure extension"() {
+        when:
+        def shouldNotExist = "shouldNotExist"
+        storage.configureExtension(shouldNotExist, {})
+        then:
+        def t = thrown UnknownDomainObjectException
+        t.message == storage.extensionWithNameDoesNotExistMessage(shouldNotExist)
```

Comment:
:x: Check the message using a constant string or regular expression instead of exposing implementation details.

Reply:
Is `package private` "exposing" an implementation detail? I've clearly marked it as `@VisibleForTesting`.

I'm happy to change it if this this is what you wish.

Reply:
It's also about having a readable specification of the message in the test.

Reply:
Will do.